File: 06_Algorithm/leetcode_2396.go
/**
 * File              : leetcode_2396.go
 * Author            : Andy
 * Date              : 2022.09.14
 * Last Modified Date: 2022.09.14
 * Last Modified By  : Andy
 */

package isStrictlyPalindromic

func convert(n, b int) int {
	rs := 0
	for n != 0 {
		rs += rs*b + n%b
		n /= b
	}
	return rs
}

func isStrictlyPalindromic(n int) bool {
	for i := 2; i < n; i++ {
		rs := convert(n, i)
		if rs != n {
			return false
		}
	}
	return true
}

File: Algorithm/leetcode_2396.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_2396.py
# Author            : Andy
# Date              : 2022.09.14
# Last Modified Date: 2022.09.14
# Last Modified By  : Andy


class Solution:

    def isStrictlyPalindromic(self, n):

        def convert(n, b):
            rs = 0
            while n:
                rs += rs * b + n % b
                n //= b
            return rs

        for i in range(2, n - 1):
            rs = convert(n, i)
            if rs != n:
                return False

        return True
